template.yaml

```diff
@@ -0,0 +1,49 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: >
+  anlessini-SAM
+
+  Sample SAM Template for anlessini-SAM
+
+# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
+Globals:
+  Function:
+    Timeout: 30
+
+Resources:
+  SearchLambdaFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: SearchLambdaFunction
+      Handler: io.anlessini.SearchLambda::handleRequest
+      Runtime: java11
+      MemorySize: 3008
+      Policies:
+        - S3ReadPolicy:
+            BucketName: '*'
+        - DynamoDBCrudPolicy:
+            TableName: '*'
+      Environment: # More info about Env Vars: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#environment-object
+        Variables:
+          INDEX_BUCKET: acl-lucene-index
```

Comment:
Is this bucket available to the public? To S3, bucket names are globally unique. If we do not want to maintain this bucket for ALL Lambda deployments using Anlessini, maybe it is worth noting to the users that they should create their own buckets and change this variable.

Reply:
Yeah makes sense, I will probably put this in the replication doc?

Reply:
IMHO we should have docs about:
1. Given a collection, how to generate Lucene index from it (prolly just point them to Anserini)
2. How to setup S3 bucket and dump index files into the bucket (can be done using Python scripts as well)
3. How to setup DynamoDB table and dump raw documents into the table (same thing as above)

Reply:
Does SAM give you the ability to provision S3 buckets and DynamoDB tables?

Reply:
If not, maybe we can also consider Terraform for describing our infrastructure in code? https://github.com/terraform-providers/terraform-provider-aws 
At this point it may be overkill, but I was under the impression that it gives you control on most AWS resources (e.g., S3, IAM, DynamoDB, Lambda & more) and it is possible to parameterize configs such as bucket name (I have seen it previously in work).

@lintool @shaneding

Reply:
Here is an example of Terraform with Lambda & DynamoDB: https://github.com/Ben-Wu/hockey-stats-grabber/tree/master/terraform, it is one of my friend's project.

Reply:
Yeah SAM does give the ability to provision S3 buckets and DynamoDB tables, I was going to look into that next. And yeah terraform will work too, although it's probably an overkill since we only use Lambda, S3 and DynamoDB